engine/suppliers.go

```diff
@@ -284,6 +285,100 @@ func (spS *SupplierService) resourceUsage(resIDs []string) (tUsage float64, err
 	return
 }
 
+func (spS *SupplierService) populateSortingData(ev *utils.CGREvent, spl *Supplier, extraOpts *optsGetSuppliers) (srtSpl *SortedSupplier, pass bool, err error) {
+	globalStats := map[string]float64{ //used for QOS strategy
+		utils.Weight: spl.Weight,
```

Comment:
the global stats here are to be used just for stats, the rest of sorting data is taken out of .SortingData